File: DevExpress-Demo/VB/PivotGridDemo.Wpf/Helpers/ServerMode/ServerModeViewModel.vb
Imports DevExpress.Xpf.DemoBase
Imports PivotGridDemo.Helpers
Imports System.Data.SQLite

Namespace PivotGridDemo

    Public Class ServerModeViewModel
        Inherits ServerModeViewModelBase(Of OrderDataRecord)

#Region "DatabaseInfo"
        Public Shared ReadOnly DatabaseInfo As DatabaseInfo = New DatabaseInfo(OrderDataContext.FileName, "OrderDataRecords", GetType(OrderDataRecord), AddressOf DatabaseHelper.CreateValues, Function(sql, connection) New SQLiteCommand(sql, CType(connection, SQLiteConnection)))

#End Region
        Protected Sub New()
            MyBase.New(DatabaseInfo, Function() New OrderDataContext().OrderDataRecords)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/PivotGridDemo.Wpf/Helpers/ServerMode/ServerModeViewModel.cs
using DevExpress.Xpf.DemoBase;
using PivotGridDemo.Helpers;
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Linq;
using System.Text;

namespace PivotGridDemo {
    public class ServerModeViewModel : ServerModeViewModelBase<OrderDataRecord> {
        #region DatabaseInfo
        public static readonly DatabaseInfo DatabaseInfo = new DatabaseInfo(
            OrderDataContext.FileName, "OrderDataRecords", typeof(OrderDataRecord), DatabaseHelper.CreateValues, (sql, connection) => new SQLiteCommand(sql, (SQLiteConnection)connection));
        #endregion

        protected ServerModeViewModel()
            : base(DatabaseInfo, () => new OrderDataContext().OrderDataRecords) {
        }
    }
}
